### PR TITLE
refactor(react-components): update side bar

### PR DESCRIPTION
**Pull request by taylrj**
This patch adds a more declarative way to configure side bar.
That is, directly declare the anchors in JSX.

### CODE REVIEW

**Comment by taylrj** [comment]
Updated.

File: packages/react-components/src/side-bar/index.js [hunk 1]
```diff
@@ -177,6 +182,17 @@ export default function decorateSideBar(DecoratedComponent) {
         modules = [children]
       }
 
+      const anchors = modules.map(module => {
+        const anchorObj = {
+          id: _.get(module.props, 'anchorId'),
+          label: _.get(module.props, 'anchorLabel', ''),
+          show: _.get(module.props, 'showAnchor', true),
```
Comment (babygoat) [inline packages/react-components/src/side-bar/index.js:189]: minor. would it be more intuitive to configure the default as `false` and display the anchor explicitly?

```
<DecoratedSideBar>
  <AnchorWrapper anchorId="id-1" anchorLabel="displayAnchor"  showAnchor >
      <SectionOne />
  </AnchorWrapper>
  <AnchorWrapper anchorId="id-2" anchorLabel="hideAnchor"  >
      <SectionTwo />
  </AnchorWrapper> 
</DecoratedSideBar>

```

**Comment by babygoat** [comment]
LGTM

File: packages/react-components/src/side-bar/index.js (hunk 1 above)
Comment (taylrj) [reply packages/react-components/src/side-bar/index.js]: Updated.

**Pull request merged**